Improbable/Postgres/Improbable.Postgres/MetricsPusher.cs

```diff
@@ -20,9 +20,16 @@ public MetricsPusher(PostgresOptions options)
 
         public void Dispose()
         {
-            StopPushingMetrics();
+            metricsCancellationTokenSource?.Cancel();
+            metricsPusherTask?.Wait();
+
             metricsCancellationTokenSource?.Dispose();
             metricsPusherTask?.Dispose();
+
+            metricsCancellationTokenSource = null;
+            metricsPusherTask = null;
+
+            GC.SuppressFinalize(this);
```

Comment:
Why are we doing this?

Reply:
It's a standard part of the `Dispose` pattern that I usually forget to do. I was feeling very C#-y yesterday and I actually remembered to do it.

Reply:
Eh, its a no-op when there isn't a finalizer defined, so I guess it can't hurt